source/fsicalmanagement/dataaccess/event_repository.d

```diff
@@ -0,0 +1,259 @@
+    @Value("mysql.table.events") string eventsTableName;
+
+public:
+    ///
+    this(MySQLPool pool)
+    {
+        this.pool = pool;
+    }
+
+    /**
+     * Saves an event to the configured MySQL table.
+     * Params:
+     * event = The `Event` to save.
+     * 
+     * Returns: The saved `Event`.
+     */
+    Event save(Event event) @trusted
+    {
+        auto cn = pool.lockConnection();
+        scope (exit)
+            cn.close;
+        auto prepared = cn.prepare("INSERT INTO " ~ eventsTableName
+                ~ " (begin, end, name, description, type, shout)" ~ " VALUES(?, ?, ?, ?, ?, ?)");
+        prepared.setArgs(event.begin, event.end, event.name, event.description,
+                event.type.to!uint, event.shout);
+        cn.exec(prepared);
+
+        return event;
+    }
+
+    /**
+     * Gets all events from the configured MySQL table.
+     *
+     * Returns: An `InputRange` containing all `Event`s from the configured
+     *          MySQL table.
+     */
+    InputRange!Event findAll() @trusted
+    {
+        import std.array : array;
+
+        auto cn = pool.lockConnection();
+        scope (exit)
+            cn.close;
+        auto prepared = cn.prepare(
+                "SELECT id, begin, end, name, description, type, shout FROM " ~ eventsTableName ~ "");
+        return cn.query(prepared).array.map!(r => toEvent(r)).inputRangeObject;
+    }
+
+    /**
+     * Gets an event by its id from the configured MySQL table.
+     * Params:
+     * id = The id of the event to get.
+     *
+     * Returns: `Nullable!Event` containing the `Event` corresponding to
+     *          $(D_PARAM id) in the configured MySQL table. or `null`, if no
+     *          such event exists.
+     */
+    Nullable!Event findById(const string id) @trusted
+    {
+        auto cn = pool.lockConnection();
+        scope (exit)
+            cn.close;
+        auto prepared = cn.prepare(
+                "SELECT id begin end name description type shout FROM "
+                ~ eventsTableName ~ " WHERE id = ?");
+        prepared.setArg(0, id.to!uint);
+        auto result = cn.query(prepared);
+
+        if (!result.empty)
+        {
+            auto event = toEvent(result.front);
+            return event.nullable;
+        }
+        return Nullable!Event.init;
+    }
+
+    /**
+     * Removes an event from the configured MySQL table.
+     * Params:
+     * id = The id of the event to remove.
+     */
+    void deleteById(const string id) @trusted
+    {
+        auto cn = pool.lockConnection();
+        scope (exit)
+            cn.close;
+        auto prepared = cn.prepare("DELETE FROM " ~ eventsTableName ~ " WHERE id = ?");
+        prepared.setArg(0, id.to!uint);
+        cn.exec(prepared);
+    }
+
+private:
+    Event toEvent(const Row r) @trusted
+    {
+        import fsicalmanagement.model.event : EventType;
+        import std.datetime.date : Date;
+
+        Event event;
+        event.id = r[0].get!uint
```

Comment:
With the sqlite driver, the returned variant has named fields, usually the with the name of the column.
Maybe this is also the case here, and can be used to increase understandability?

Reply:
The `Row` only supports two ways to extract the data:
* `opIndex`, which returns a `Variant` (this is what we are currently using)
* `toStruct`, which converts the row to a compatible struct. Unfortunately, our `Event` struct is not directly compatible (mysql-native can not handle `Nullable` types at the moment)

Named access to fields is not possible at the moment.